doc/source/release-notes/Percona-Server-5.7.12-5.rst

```diff
@@ -0,0 +1,121 @@
+ to server crash. Bug fixed :bug:`1577282`.
+
+ |InnoDB| crash recovery might fail if :variable:`innodb_flush_method` was set
+ to ``ALL_O_DIRECT``. Bug fixed :bug:`1529885`.
+
+ Fixed heap allocator/deallocator mismatch in
+ :ref:`scalability_metrics_plugin`. Bug fixed :bug:`1581051`.
+
+ |Percona Server| is now built with system ``zlib`` library instead of the
+ older bundled one. Bug fixed :bug:`1108016`.
+
+ ``CMake`` would fail if |TokuDB| tests passed. Bug fixed :bug:`1521566`.
+
+ Reduced the memory overhead per page in the |InnoDB| buffer pool. The fix was
+ based on Facebook patch
+ `#91e979e <https://github.com/facebook/mysql-5.6/commit/91e979e8436b83400e918fa0f251036e50d0cb5f>`_.
+ Bug fixed :bug:`1536693` (upstream :mysqlbug:`72466`).
+
+ ``CREATE TABLE ... LIKE ...`` could create a system table with an unsupported
+ enforced engine. Bug fixed :bug:`1540338`.
+
+ Change buffer merge could throttle to 5% of I/O capacity on an idle server.
+ Bug fixed :bug:`1547525`.
+
+ Parallel doublewrite memory was not freed with
+ :variable:`innodb_fast_shutdown` was set to ``2``. Bug fixed :bug:`1578139`.
+
+ Server will now show more descriptive error message when |Percona Server|
+ fails with ``errno == 22 "Invalid argument"``, if
+ :variable:`innodb_flush_method` was set to ``ALL_O_DIRECT``. Bug fixed
+ :bug:`1578604`.
+
+ The error log warning ``Too many connections`` was only printed for connection
+ attempts when :variable:`max_connections` + one ``SUPER`` have connected. If
+ the extra ``SUPER`` is not connected, the warning was not printed for a
+ non-SUPER connection attempt. Bug fixed :bug:`1583553`.
+
+ ``apt-cache show`` command for ``percona-server-client`` was showing
+ ``innotop`` included as part of the package. Bug fixed :bug:`1201074`.
+
+ Replication didn't work when ``@@GLOBAL.SERVER_UUID`` was missing on the
+ master. Bug fixed :bug:`1566642` (upstream :mysqlbug:`80962`).
+
+ Upgrade logic for figuring if |TokuDB| upgrade can be performed from the
+ version on disk to the current version was broken due to regression introduced
+ when fixing :ftbug:`684` in |Percona Server| :rn:`5.7.11-4`. Bug fixed
+ :ftbug:`717`.
+
+ Fixed ``jemalloc`` version parsing error. Bug fixed :tokubug:`528`.
+
+ If ``ALTER TABLE`` was run while :variable:`tokudb_auto_analyze` variable was
+ enabled it would trigger auto-analysis, which could lead to a server crash if
+ ``ALTER TABLE DROP KEY`` was used because it would be operating on the old
+ table/key meta-data. Bug fixed :tokubug:`945`.
+
+ The |TokuDB| storage engine with :variable:`tokudb_pk_insert_mode` set to
+ ``1`` is safe to use in all conditions. On ``INSERT IGNORE`` or ``REPLACE
+ INTO``, it tests to see if triggers exist on the table, or replication is
+ active with ``!BINLOG_FORMAT_STMT`` before it allows the optimization. If
+ either of these conditions are met, then it falls back to the "safe" operation
+ of looking up the target row first. Bug fixed :tokubug:`952`.
+
```

Comment:
Please see https://tokutek.atlassian.net/browse/DB-952 to see that pk_insert_mode is deprecated in 5.7